tests/unit/services/apigateway/test_handler_integration_request.py

```diff
@@ -0,0 +1,265 @@
+)
+from localstack.services.apigateway.next_gen.execute_api.gateway_response import (
+    UnsupportedMediaTypeError,
+)
+from localstack.services.apigateway.next_gen.execute_api.handlers import IntegrationRequestHandler
+from localstack.services.apigateway.next_gen.execute_api.handlers.integration_request import (
+    PassthroughBehavior,
+)
+from localstack.services.apigateway.next_gen.execute_api.variables import (
+    ContextVariables,
+    ContextVarsRequestOverride,
+)
+from localstack.testing.config import TEST_AWS_ACCOUNT_ID, TEST_AWS_REGION_NAME
+
+TEST_API_ID = "test-api"
+TEST_API_STAGE = "stage"
+
+
+@pytest.fixture
+def create_context():
+    """
+    Create a context populated with what we would expect to receive from the chain at runtime.
+    We assume that the parser and other handler have successfully populated the context to this point.
+    """
+
+    def _create_context(
+        method: Method = None,
+        request: InvocationRequest = None,
+    ):
+        context = RestApiInvocationContext(Request())
+
+        # The api key validator only relies on the raw headers from the invocation requests
+        context.invocation_request = request or InvocationRequest()
+
+        # Frozen deployment populated by the router
+        context.deployment = RestApiDeployment(
+            account_id=TEST_AWS_ACCOUNT_ID,
+            region=TEST_AWS_REGION_NAME,
+            rest_api=MergedRestApi(rest_api={}),
+        )
+
+        # Context populated by parser handler
+        context.region = TEST_AWS_REGION_NAME
+        context.account_id = TEST_AWS_ACCOUNT_ID
+        context.stage = TEST_API_STAGE
+        context.api_id = TEST_API_ID
+        context.resource_method = method or Method(
+            methodIntegration=Integration(
+                type=IntegrationType.HTTP,
+                requestParameters=None,
+                uri="https://example.com",
+                httpMethod="POST",
+            )
+        )
+        context.context_variables = ContextVariables(
+            resourceId="resource-id",
+            apiId=TEST_API_ID,
+            httpMethod="POST",
+            path="/stage/root/{path}",
+            requestOverride=ContextVarsRequestOverride(header={}, path={}, querystring={}),
+            resourcePath="/root/{path}",
+            stage="stage",
+        )
+
+        return context
+
+    return _create_context
+
+
+@pytest.fixture
+def default_invocation_request() -> InvocationRequest:
+    header_dict = {"header": ["ignored-value", "test-header-value"]}
+    headers = Headers(header_dict)
+    return InvocationRequest(
+        http_method=HTTPMethod.POST,
+        raw_path="/test/test-path-value",
+        path="/test/test-path-value",
+        path_parameters={"path": "test-path-value"},
+        query_string_parameters={"qs": "test-qs-value"},
+        raw_headers=headers,
+        headers=dict(headers),
+        multi_value_query_string_parameters={"qs": ["ignored-value", "test-qs-value"]},
```

Comment:
I believe in AWS, in their context, if you have `multi_value_query_string_parameters={"qs": ["val1", "val2"]}` then `query_string_parameters={"qs": "val1"}`. It's really weird, but it is how it is. 

I wonder if should try to use the parse part of the handler here for a request, that might avoid duplication of the headers and such? We could try refactoring previous tests with it and see how that goes. 
We could manually invoke `InvocationRequestParser().create_invocation_request`.

Reply:
That is a godd idea! Let me see what I can do